src/components/QuranReader/SidebarNavigation/SidebarNavigation.tsx

```diff
@@ -0,0 +1,119 @@
+import Switch from 'src/components/dls/Switch/Switch';
+import useOutsideClickDetector from 'src/hooks/useOutsideClickDetector';
+import { selectContextMenu } from 'src/redux/slices/QuranReader/contextMenu';
+import {
+  selectIsSidebarNavigationVisible,
+  selectNavigationItem,
+  selectSelectedNavigationItem,
+  NavigationItem,
+  setIsVisible,
+} from 'src/redux/slices/QuranReader/sidebarNavigation';
+
+const Loading = () => (
+  <div className={styles.loadingContainer}>
+    <Spinner />
+  </div>
+);
+
+const PageSelection = dynamic(() => import('./PageSelection'), { loading: Loading });
+const SurahSelection = dynamic(() => import('./SurahSelection'), { loading: Loading });
+const JuzSelection = dynamic(() => import('./JuzSelection'), {
+  loading: Loading,
+});
+
+const SidebarNavigation = () => {
+  const { isExpanded: isContextMenuExpanded } = useSelector(selectContextMenu, shallowEqual);
+  const isVisible = useSelector(selectIsSidebarNavigationVisible);
+  const selectedNavigationItem = useSelector(selectSelectedNavigationItem);
+  const dispatch = useDispatch();
+  const { t } = useTranslation('common');
+  const sidebarRef = useRef();
+
+  useOutsideClickDetector(
+    sidebarRef,
+    () => {
+      dispatch(setIsVisible(false));
+    },
+    isMobile(),
+  );
+
+  const navigationItems = [
+    {
+      name: t('surah'),
+      value: NavigationItem.Surah,
+    },
+    {
+      name: t('juz'),
+      value: NavigationItem.Juz,
+    },
+    {
+      name: t('page'),
+      value: NavigationItem.Page,
+    },
+  ];
+
+  return (
+    <div
+      ref={sidebarRef}
+      className={classNames(styles.container, {
+        [styles.visibleContainer]: isVisible,
+        [styles.spaceOnTop]: isContextMenuExpanded,
+      })}
+    >
+      <div className={styles.header}>
+        <div className={styles.switchContainer}>
+          <Switch
+            items={navigationItems}
+            selected={selectedNavigationItem}
+            onSelect={(value) => {
+              dispatch(selectNavigationItem(value as NavigationItem));
+            }}
+          />
+        </div>
+        <Button
+          tooltip={t('close')}
+          shape={ButtonShape.Circle}
+          size={ButtonSize.Small}
+          variant={ButtonVariant.Ghost}
+          onClick={() => {
+            dispatch(setIsVisible(false));
+          }}
+        >
+          <IconClose />
+        </Button>
+      </div>
+      <p className={styles.tip}>
+        <span>{t('sidebar.try-navigating-with')}</span>
+        <KeyboardInput meta keyboardKey="K" />
+      </p>
+      <div className={styles.contentContainer}>
+        {selectedNavigationItem === NavigationItem.Surah && <SurahSelection />}
+        {selectedNavigationItem === NavigationItem.Juz && <JuzSelection />}
+        {selectedNavigationItem === NavigationItem.Page && <PageSelection />}
+      </div>
+    </div>
+  );
+};
+
+const TABLET_WIDTH = 768;
+const isMobile = () => {
```

Comment:
Nothing actionable now but if we needed this logic again in the future, we should move it to a util.